### PR TITLE
OpenTracingSpringMvcAutoConfiguration add ConditionalOnClass SpringMvcSofaTracerFilter

### PR DESCRIPTION
**Pull request by caojie09**
OpenTracingSpringMvcAutoConfiguration add ConditionalOnClass SpringMvcSofaTracerFilter. #111

### CODE REVIEW

**Comment by khotyn** [comment]
Ant testcase?

**Comment by caojie09** [comment]
@khotyn I want to make OpenTracingSpringMvcAutoConfiguration invalid, there are two ways to achieve: 

* add @EnableAutoConfiguration(exclude = {OpenTracingSpringMvcAutoConfiguration.class})
* remove sofa-tracer-springmvc-plugin

The first solution does not take effect before 2.2.0 becase of [ComponentScan](https://github.com/alipay/sofa-tracer/issues/110) 
The second solution does not take effect  in all the version, I think it should be fixed.

**Comment by khotyn** [comment]
@caojie09 可以加测试用例吗？

File: tracer-sofa-boot-starter/src/main/java/com/alipay/sofa/tracer/boot/springmvc/configuration/OpenTracingSpringMvcAutoConfiguration.java [hunk 1]
```diff
@@ -37,6 +38,7 @@
  */
 @Configuration
 @EnableConfigurationProperties(OpenTracingSpringMvcProperties.class)
+@ConditionalOnClass(SpringMvcSofaTracerFilter.class)
```
Comment (QilongZhang) [inline tracer-sofa-boot-starter/src/main/java/com/alipay/sofa/tracer/boot/springmvc/configuration/OpenTracingSpringMvcAutoConfiguration.java]: 建议加在方法上，在3.x版本中，这个Configuration中不止这一个Bean，避免cherry-pick修改代码。

**Pull request merged**